session/session.go

```diff
@@ -90,6 +108,152 @@ func Open(config sfdc.Configuration) (*Session, error) {
 	return session, nil
 }
 
+func openDeviceSession(config sfdc.Configuration) (*Session, error) {
+	request, err := buildDeviceAuthenticationFlowInitiationRequest(config.Credentials)
+	if err != nil {
+		return nil, fmt.Errorf("failed to build ")
+	}
+
+	initResp, err := makeDeviceAuthenticationFlowInitiationRequest(request, config.Client)
+	if err != nil {
+		return nil, err
+	}
+
+	var tokenResp *accessTokenResponse
+	var done = make(chan struct{})
+
+	go func() {
```

Comment:
I think we can avoid this goroutine  and simplify this block by using a [time.Ticker](https://golang.org/pkg/time/#example_NewTicker), something like:
```go
ticker := time.NewTicker(time.Duration(initResp.Interval+1) * time.Second)
defer ticker.Stop()

for range ticker.C {
// ...
}
```

Proof of concept here: https://play.golang.org/p/QbvIE_3wNIx